Fib-Board/lib/leaderboard.h

```diff
@@ -0,0 +1,106 @@
+#include "filehandling.h"
+
+void swap(player *p1,player *p2)
+{
+    player *temp;
+    *temp=*p1;
+    *p1=*p2;
+    *p2=*temp;
+}
+
+void sortLeaderBoard(int n,player players[n])
+{
+   int i, j;  
+    for (i = 0; i < n-1; i++)      
+    { 
+        for (j = 0; j < n-i-1; j++)  
+            if ((players[j].noOfMoves) > (players[j+1].noOfMoves))
+            {
+                swap(&players[j], &players[j+1]);
+            }
+    } 
+}
+
+int getLeaderBoard(player players[])
+{
+    FILE *infile;
+    infile = fopen ("../data/LeaderBoard.dat", "r"); 
+    if (infile == NULL) 
+    { 
+        fprintf(stderr, "\nError opening file\n"); 
+        exit (1); 
+    }
+    int index=0;
+    while(fread(&players[index], sizeof(player), 1, infile)!=0)
+    {
+       //printf("\nname:%s",players[index].name);
+        index++;
+    } 
+    return index;
+}
+
+void writeIntoLeadeBoard(int n,player players[n])
+{
+    FILE *outfile; 
+    outfile = fopen ("../data/LeaderBoard.dat", "w");
+    if (outfile == NULL) 
+    { 
+        fprintf(stderr, "\nError opend file\n"); 
+        exit (1); 
+    } 
+    int i=0;
+    for(i=0;i<n;i++)
+         fwrite (&players[i], sizeof(player), 1, outfile);

```

Comment:
Here I am writing structure into the file but while writing, there are also some garbage structures being stored. 

Ex: if this loop runs for 1 time we expect only one state of structure to be written into the file but that's not happening. Along with the structure we expect, there are also some garbage structures been stored in the file.

I hope you  understood. 😅